### PR TITLE
Fcp 569

### PR DESCRIPTION
**Pull request by OlivierDelalande**
getting acr_values from req.sessions.infos.acr_values to set identity provider login page dropdown to this acr_values

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -8,6 +8,23 @@ const messages = {
...
   },
 };
+const DEFAULT_EIDAS_LEVEL = 'eidas3';
+
+export const getEidasLevel = (req) => {
+  if (!req.session || !req.session.infos || !req.session.infos.acr_values) {
+    return DEFAULT_EIDAS_LEVEL;
+  }
+
+  const validEidasLevel = ['eidas1', 'eidas2', 'eidas3'];
```
Comment (g4vroche) [inline src/router.js]: This constant could be defined outside of the function, and in caps, like  `DEFAULT_EIDAS_LEVEL` is;

Those two constant might also be stored somewhere else in the project, maybe in a configuration file to allow easy tweak of this parameter (I'm not aware of a real world use case, though).

File: src/router.js [hunk 2]
```diff
@@ -18,7 +35,9 @@ export const mountRoutes = (app, provider) => {
 
   app.use('/interaction', urlencoded({ extended: false }));
 
-  app.get('/', (req, res) => { res.sendStatus(200); });
+  app.get('/', (req, res) => {
```
Comment (g4vroche) [inline src/router.js]: We should check with @rdubigny about code style before submitting automated formatting changes.

We might need a `.prettierrc` on this project, unless it is supposedly up to eslint to handle formatting ?

File: src/views/sign-in.ejs [hunk 3]
```diff
@@ -27,9 +27,9 @@
                 <a title="Forcer le niveau eIDAS renvoyé par le Fournisseur d'Identité ne fonctionnera que si le Fournisseur de Service a précisé le paramètre &quot;acr_values&quot; dans l'appel à FranceConnect. Le cas échéant, le niveau faible (eidas1) sera renvoyé par FranceConnect">(info)</a>
             </label>
             <select class="form-control" required name="acr" placeholder="Niveau eIDAS (à renvoyer à FC)*">
-                <option value="eidas1" selected>Faible (eidas1)</option>
-                <option value="eidas2">Substantiel (eidas2)</option>
-                <option value="eidas3">Élevé (eidas3)</option>
+                <option value="eidas1" <%= !acr || acr && acr == 'eidas1' ? 'selected' : '' %>>Faible (eidas1)</option>
```
Comment (g4vroche) [inline src/views/sign-in.ejs]: We can be confident about `acr`being defined with the `getEidasLevel()` function, so we can skip the boolean test.

Moreover, default level is `eidas3`, not `eidas1`

File: src/router.js [hunk 4]
```diff
@@ -8,6 +8,23 @@ const messages = {
     message: 'Email ou mot de passe incorrect.',
   },
 };
+const DEFAULT_EIDAS_LEVEL = 'eidas3';
+
+export const getEidasLevel = (req) => {
+  if (!req.session || !req.session.infos || !req.session.infos.acr_values) {
```
Comment (rdubigny) [inline src/router.js]: could plz tell me where this session come from ? :tea: I think reading the query params should be enough

File: src/router.js [hunk 5]
```diff
@@ -27,6 +28,11 @@ export const mountRoutes = (app, provider) => {
         interaction: { error, error_description: errorDescription },
```
Comment (rdubigny) [inline src/router.js]: the params you are looking for is here:
```
params: { acr_values }
```

File: src/views/sign-in.ejs [hunk 6]
```diff
@@ -24,12 +24,12 @@
 
         <div class="form__group">
             <label for="acr">Niveau eIDAS
-                <a title="Forcer le niveau eIDAS renvoyé par le Fournisseur d'Identité ne fonctionnera que si le Fournisseur de Service a précisé le paramètre &quot;acr_values&quot; dans l'appel à FranceConnect. Le cas échéant, le niveau faible (eidas1) sera renvoyé par FranceConnect">(info)</a>
+                <a title="Forcer le niveau eIDAS renvoyé par le Fournisseur d'Identité ne fonctionnera que si le Fournisseur de Service a précisé le paramètre &quot;acr&quot; dans l'appel à FranceConnect.">(info)</a>
```
Comment (rdubigny) [inline src/views/sign-in.ejs]: acr => acr_values

we are talking about the query param send by FC

File: src/router.js [hunk 7]
```diff
@@ -67,14 +74,15 @@ export const mountRoutes = (app, provider) => {
       return await provider.interactionFinished(req, res, result);
     } catch (error) {
       if (error.message === 'invalid_credentials') {
-        return res.redirect(`/interaction/${req.params.grant}?notification=${error.message}`);
+        return res.redirect(
```
Comment (rdubigny) [inline src/router.js]: lenght is 95, no need to lint this

**Pull request merged**